node/builder_chain.go

```diff
@@ -79,17 +79,36 @@ var ChainNode = Options(
 	Override(new(store.WeightFunc), filcns.Weight),
 	Override(new(stmgr.Executor), consensus.NewTipSetExecutor(filcns.RewardFunc)),
 	Override(new(consensus.Consensus), filcns.NewFilecoinExpectedConsensus),
-	Override(new(*store.ChainStore), modules.ChainStore),
+
+	If(isNotChainCassandra(),
+		Override(new(*store.ChainStore), modules.ChainStore),
+	),
+	If(isChainCassandra(),
+		ApplyIf(isNotFollower,
+			Override(new(*store.ChainStore), modules.CassandraMetadataChainStore),
+		),
+
+		ApplyIf(isFollower,
+			Override(new(*store.ChainStore), modules.CassandraReadonlyMetadataChainStore),
+		),
+	),
 	Override(new(*stmgr.StateManager), modules.StateManager),
 	Override(new(dtypes.ChainBitswap), modules.ChainBitswap),
 	Override(new(dtypes.ChainBlockService), modules.ChainBlockService), // todo: unused
 
 	// Consensus: Chain sync
-
 	// We don't want the SyncManagerCtor to be used as an fx constructor, but rather as a value.
 	// It will be called implicitly by the Syncer constructor.
 	Override(new(chain.SyncManagerCtor), func() chain.SyncManagerCtor { return chain.NewSyncManager }),
-	Override(new(*chain.Syncer), modules.NewSyncer),
+
+	ApplyIf(isNotFollower,
+		Override(new(*chain.Syncer), modules.NewSyncer),
+	),
+
+	ApplyIf(isFollower,
+		Override(new(*chain.Syncer), modules.NewSyncerFollower),
```

Comment:
In the follower mode we shouldn't need any of:
* `chain.SyncManagerCtor`
* `*chain.Syncer`
* `*hello.Service` / `RunHelloKey` / `HandleIncomingBlocksKey`
* `*peermgr.PeerMgr`
* `exchange.Server`
* Possibly should also somehow disable pubsub subscriptions

Techincally this is not critical for landing this PR in a very experimental form, but not running those services would be much more correct

Reply:
resolved in node/builder_chain.go